assets/app/scripts/directives/metrics.js

```diff
@@ -0,0 +1,257 @@
+                format: '%H:%M'
+              }
+            },
+            y: {
+              label: {
+                position: 'outer-center'
+              },
+              min: 0,
+              // With default padding you can have negative axis tick values.
+              padding: {
+                left: 0,
+                top: 0,
+                bottom: 0
+              },
+              show: true,
+              tick: {
+                count: 5,
+                fit: true,
+                format: function(value) {
+                  return d3.round(value, 1);
+                }
+              }
+            }
+          }
+        };
+
+        function updateChart(data, metric) {
+          var sparklineConfig,
+              container = scope.options.selectedContainer,
+              convertUnitsFn = convertUnitsFnByMetric[metric.id];
+          var chartData = scope.chartDataByMetric[metric.id] = {
+            xData: ['dates'],
+            yData: [metric.units]
+          };
+
+          // The patternfly charts change the config objects we pass on
+          // initialization, so just update what's there rather than recreating
+          // the object if it exists.
+          if (!scope.utilizationConfigByMetric[metric.id]) {
+            scope.utilizationConfigByMetric[metric.id] = {
+              units: metric.units
+            };
+          }
+
+          if (!scope.donutConfigByMetric[metric.id]) {
+            scope.donutConfigByMetric[metric.id] = {
+              units: metric.units,
+              chartId: metric.chartId
+            };
+          }
+
+          if (!scope.sparklineConfigByMetric[metric.id]) {
+            // Make a copy of the base sparkline config to edit it.
+            scope.sparklineConfigByMetric[metric.id] = angular.copy(baseSparklineConfig);
+          }
+          sparklineConfig = scope.sparklineConfigByMetric[metric.id];
+          sparklineConfig.chartId = metric.chartId;
+          sparklineConfig.units = metric.units;
+          sparklineConfig.axis.y.label = metric.units;
+
+          // If we're showing data from another day, add the abbreviated day name
+          // to the time format.
+          var start = moment().subtract(scope.options.timeRange.value, 'minutes');
+          if (start.isSame(moment(), 'day')) {
+            sparklineConfig.axis.x.tick.format = '%H:%M';
+          } else {
+            sparklineConfig.axis.x.tick.format = '%a %H:%M';
+          }
+
+          var limit;
+          switch (metric.id) {
+          case 'memory/usage':
+            var memLimit = getMemoryLimit(container);
+            if (memLimit) {
+              // Convert to MB. usageValueFilter returns bytes.
+              limit = bytesToMB(usageValueFilter(memLimit));
+            }
+            break;
+          case 'cpu/usage':
+            var cpuLimit = getCPULimit(container);
+            if (cpuLimit) {
+              // Remove the 'm' unit at the end.
```

Comment:
It will strip the m off. It has to do that for limit ranges on the settings
page.
On Oct 12, 2015 9:26 AM, "Sam Padgett" <notifications@github.com> wrote:

> In assets/app/scripts/directives/metrics.js
> <https://github.com/openshift/origin/pull/4999#discussion_r41755497>:
>
> > +            sparklineConfig.axis.x.tick.format = '%a %H:%M';
> > +          }
> > +
> > +          var limit;
> > +          switch (metric.id) {
> > +          case 'memory/usage':
> > +            var memLimit = getMemoryLimit(container);
> > +            if (memLimit) {
> > +              // Convert to MB. usageValueFilter returns bytes.
> > +              limit = bytesToMB(usageValueFilter(memLimit));
> > +            }
> > +            break;
> > +          case 'cpu/usage':
> > +            var cpuLimit = getCPULimit(container);
> > +            if (cpuLimit) {
> > +              // Remove the 'm' unit at the end.
>
> Yeah, this CPU limit code is not complete. Does usageValueFilter deal with
> millicores?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4999/files#r41755497>.
>